### DISCUSSION TITLE
Panel plugin not found: grafana-map-panel

### DISCUSSION THREAD

**Discussion by withanhdammit** [teslamate-org/teslamate#1901] [Getting started]
**Describe the bug**

Charging Stats panel reports "Panel plugin not found: grafana-map-panel"
Extensive searching has not found where to download this panel

**Expected behavior**

Display map panel

**How to reproduce it (as minimally and precisely as possible)**:

Install using the Manual Install instructions.

**Screenshots**

![image](https://user-images.githubusercontent.com/16404870/128640565-1aba0568-d59a-44cb-9c66-6365ec235044.png)

**Environment**

- TeslaMate version: 1.23.7
- Type of installation: Manual
- OS TeslaMate is installed on: Debain 10 Buster
- User OS & Browser: Windows 10, Chrome

**Comment by miguelangelnieto** [comment]
Hello,

Try the following one:

`grafana-cli  --pluginUrl https://github.com/panodata/grafana-map-panel/releases/download/0.15.0/grafana-map-panel-0.15.0.zip plugins install grafana-worldmap-panel-ng`

**Reply by iandees** [reply]
This is successful:

```
$ sudo grafana-cli --pluginUrl https://github.com/panodata/grafana-map-panel/releases/download/0.15.0/grafana-map-panel-0.15.0.zip plugins install grafana-worldmap-panel-ng
[sudo] password for iandees:
✔ Downloaded grafana-map-panel v0.15.0 zip successfully

Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.

$ sudo systemctl restart grafana-server
```

but the missing plugin error remains. I'm on Grafana 8.2.2

**Reply by iandees** [reply]
Downgrading to Grafana 7.5.10 (as suggested over in https://github.com/adriankumpf/teslamate/discussions/2000) seems to have helped here.

**Comment by withanhdammit** [comment]
Downgrading to Grafana 7.5.11 worked perfectly for me, I also had to allow the unsigned grafana-map-panel.  This totally fixed all the issues I was having!

**Comment by withinfocus** [comment]
A hopefully-helpful note: with TeslaMate 1.26+ and Grafana 8+ you'll now use the `panodata-map-panel` plugin v0.16+. Since it's unsigned and installed from a URL vs. the Grafana directory you'll need to place:

`allow_loading_unsigned_plugins = panodata-map-panel`

in your `grafana.ini` and restart Grafana to get that plugin to load.

**Comment by manuelkamp** [comment]
Hi,

I added allow_loading_unsigned_plugins = panodata-map-panel to grafana.ini and restarted grafana. Then installed
sudo grafana-cli --pluginUrl https://github.com/panodata/panodata-map-panel/releases/download/0.16.0/panodata-map-panel-0.16.0.zip plugins install grafana-worldmap-panel-ng

but I still get the error...?

**Reply by jmalcaide** [reply]
Hello,

This issue only affects manual installations. TeslaMate's Grafana Docker container includes the necessary definitions of environment variables for installing unsigned plugins. Since I use a manual installation on FreeBSD, I had to edit grafana.ini to solve this problem.

**Reply by manuelkamp** [reply]
yes, I have manual installation on proxmox container. Can not switch to docker or freebsd. is there another change in the ini file except the one i already did?

**Reply by hmin03** [reply]
@manuelkamp Did you remove the semicolon in front of the configuration line for allow_loading_unsigned_plugins in `/etc/grafana/grafana.ini`?

To override the setting in default.ini, you need to remove the semicolon.

editing reason: prevent editing the default.ini

**Comment by brianmay** [comment]
I had this problem, but the solution in https://github.com/adriankumpf/teslamate/discussions/1901#discussioncomment-2053804 fixed it for me.

@manuelkamp I think you got the command wrong, the correct command is:

```
grafana-cli --pluginUrl https://github.com/panodata/panodata-map-panel/releases/download/0.16.0/panodata-map-panel-0.16.0.zip plugins install panodata-map-panel
```

You need to install panodata-map-panel not grafana-worldmap-panel-ng

**Reply by manuelkamp** [reply]
Thanks but: this command complains about: "Error: Incorrect Usage: flag provided but not defined: -v"

If I run the command without the -v flag, it says "Downloaded...successfully". But after Grafana restart it still does not work... Idk..

**Comment by brianmay** [comment]
Sorry, don't use the -v. I deleted it from the previous post. That sounds exactly like the symptoms I got before I set `allow_loading_unsigned_plugins = panodata-map-panel`. grafana silently ignores the plugin because it isn't signed.

**Reply by manuelkamp** [reply]
yes, I already had that set in the ini since the previous install of the wrong (as you told me) plugin. It is not working for me so far... I guess this is a big mess with this plugin with different names and I am curious how many also do not get it to work and just live with it... Who came up with this :D

anyway, thanks for help!

edit: it may be that the position where I put it in the INI is wrong. since I had no sample entry of it, i just put it at the end of the file? Is that correct, or does it neet to be on a specific position?

**Reply by manuelkamp** [reply]
alright, i just found out, that you need to define also a section [plugins] before the line for allow_loading_unsigned_plugins if you do not have one in the ini file. if you have this section, you need to put your line here, and nowhere else - then it works!

found here: https://grafana.com/docs/grafana/latest/administration/configuration/#plugins-1

**Reply by brianmay** [reply]
Come to think about it, the instructions were vague - a pet hate of mine, and I missed an opportunity to clarify them. Apologies for that. Just to clarify for anybody reading this at a later date:

```
grafana-cli --pluginUrl https://github.com/panodata/panodata-map-panel/releases/download/0.16.0/panodata-map-panel-0.16.0.zip plugins install panodata-map-panel
```

And edit the `grafana.ini` (sorry, I can't help with this location, likely to vary depending on your install):

```
[plugins] 
allow_loading_unsigned_plugins = panodata-map-panel
```

And as per above, this is documented at https://grafana.com/docs/grafana/latest/administration/configuration/#plugins-1

**Comment by brianmay** [comment]
Personally I think maybe teslamate shouldn't be relying on an unsigned grafana plugin like this. And I suspect it might be a sign that the plugin isn't being maintained any more.  Looking at the git commit log it seems like it hasn't had any significant updates for some time:

https://github.com/panodata/panodata-map-panel/commits/main

Perhaps we should open a bug report asking that references to *panodata-map-panel* be replaced by a supported plugin that is being maintained? Maybe grafana-worldmap-panel?

**Comment by driehle** [comment]
Following panodata/panodata-map-panel#108 and panodata/panodata-map-panel#112, I found that Grafana 8.0 has a `geomap` plugin. With a bit of configuration, this one works actually quite well for the above-mentioned panel. I changed the plugin type to "Geomap", did a bit of configuration (mostly Data: Query A, Styles Size: chg_total, Styles: Color: chg_total) and the result is actually quite good, I'd say:

![grafik](https://github.com/adriankumpf/teslamate/assets/1586788/bed11071-b307-4da4-81d3-34c48c004ef7)

I am not sure how the original `panodata-map-panel` would have looked like, since I never used it. However, since this is quite informative, it could probably be a new default configuration?

**Reply by swiffer** [reply]
routes functionality had been introduced as alpha in grafana 9 - therefore replacing it is not a vaid option right now. I wonder if there are any plans on upgrading the bundled grafana version. I'm using Grafana 10 externally which deprecates angularjs plugin support. therefore looking for a replacement and geomap works great in my case!

https://github.com/grafana/grafana/pull/43554

**Reply by swiffer** [reply]
see #3382 - great work on getting to grafana latest
